ipa-core/src/ff/boolean_array.rs

```diff
@@ -377,3 +389,33 @@ boolean_array_impl!(
         0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0
     ]
 );
+
+impl From<(u128, u128)> for BA256 {
+    fn from(value: (u128, u128)) -> Self {
+        let iter = value
+            .0
+            .to_le_bytes()
+            .into_iter()
+            .chain(value.1.to_le_bytes());
+        let arr = GenericArray::<u8, U32>::try_from_iter(iter).unwrap();
+        BA256::deserialize(&arr)
+    }
+}
+
+// TODO: don't generate the default impls; enable these instead.
+/*
+impl FromPrss for (BA256, BA256) {
+    fn from_prss<P: SharedRandomness + ?Sized, I: Into<u128>>(prss: &P, index: I) -> (BA256, BA256) {
+        let index = index.into();
+        let (l0, r0) = prss.generate_values(2 * index);
+        let (l1, r1) = prss.generate_values(2 * index + 1);
+        (BA256::from((l0, l1)), BA256::from((r0, r1)))
+    }
+}
+
+impl rand::distributions::Distribution<BA256> for rand::distributions::Standard {
+    fn sample<R: crate::rand::Rng + ?Sized>(&self, rng: &mut R) -> BA256 {
+        BA256::from((rng.gen(), rng.gen()))
+    }
+}
+*/
```

Comment:
Do you want to land these comments? Or remove them before landing?

Reply:
These I want to keep -- I believe this is the eventual form of what they should look like for BA256. One of the changes I tried to make before realizing that it's hard to remove `Field` from the BooleanArrays right now was to have the macro-generated boolean array implementation only generate its version of these for sizes up to 128 bits. Unfortunately, needing to retain Field means needing to retain FromRandomU128, which has its own FromPrss that conflicts with this one.

Reply:
Do you plan to implement the trait from PRSS for any type that needs an MPC multiplication?

Reply:
That was my plan. Does that seem like a problem?

Reply:
isn't that reimplementing Distribution<Standard> with the only difference using a seed instead of a RNG? We could also consider having PRSS returning an RNG (and implement an RNG that panics when we request too many bits). This would allow us to use the standard rust conventions.

Reply:
Here with panic, I mean debug panic such that we could identify when we requesting AES evaluations for identical inputs twice.